app/integration/runtime/src/main/java/io/syndesis/integration/runtime/capture/OutMessageCaptureProcessor.java

```diff
@@ -44,6 +44,13 @@ public void process(Exchange exchange) throws Exception {
                 ((MessageSupport) copy).setExchange(message.getExchange());
             }
 
+            //If it is a stream cache, it can only be read once
+            //We are going to read it more than once
+            //to reuse variables from previous steps
+            if(copy.getBody() instanceof InputStreamCache) {
+                copy.setBody(copy.getBody(String.class));
```

Comment:
Bodies could be quite large streams, say files from S3. Or even binary and converting to and from Strings might corrupt them.